src/decisionengine_modules/NERSC/sources/NerscSFApi.py

```diff
@@ -0,0 +1,146 @@
+# SPDX-FileCopyrightText: 2017 Fermi Research Alliance, LLC
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Get allocation info from Nersc SuperFacilityAPI
+"""
+import json
+
+# import time
+import os
+
+import jwt
+import pandas as pd
+import pem
+import requests
+
+# from decisionengine_modules.NERSC.util import newt
+# from datetime import datetime
+from authlib.integrations.requests_client import OAuth2Session
+from authlib.oauth2.rfc7523 import PrivateKeyJWT
+
+from decisionengine.framework.modules import Source
+from decisionengine.framework.modules.Source import Parameter
+
+
+@Source.supports_config(Parameter("constraints", type=dict, comment="""Supports the layout:"""))
+@Source.produces(Nersc_Allocation_SFAPI=pd.DataFrame)
+class NerscSFApi(Source.Source):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.constraints = config.get("constraints")
+        if not isinstance(self.constraints, dict):
+            raise RuntimeError("constraints should be a dict")
+
+        self.logger = self.logger.bind(
+            class_module=__name__.split(".")[-1],
+        )
+        self.localmap = {"uscms": "m2612", "fife": "m3249"}
+        self.keys_list = ["hours_given", "hours_used", "id", "project_hours_given", "project_hours_used", "repo_name"]
+
+    def check_accesstoken(self, nersc_user):
+        token_url = "https://oidc.nersc.gov/c2id/token"
+        #        currenttime = time.time()
+        renew_bool = False
+
+        rawfile_ucms = "/tmp/ucms_access.token"
+        rawfile_fife = "/tmp/fife_access.token"
```

Comment:
These seem a cache for tokens. They should be probably in a temp directory, protected from others to read, with a name depending on the user name (see the other comment about not hardcoding user names)

Reply:
Well, I am not sure if making changes to these lines will achieve any practical gain
because these tokens do not have any write privileges to our resources in Nersc (i.e. they are read only)
I think /tmp/ might be sufficient for the location of the tokens because if the tokens are deleted
the code will generate new ones anyway..
So, let me leave them as they are and update later if we see a stronger reason for a change.

Reply:
This has also been moved to the config file.